components/org.wso2.is.configuration.toml.generator/src/main/java/org/wso2/is/configuration/toml/generator/util/TomlGenerator.java

```diff
@@ -0,0 +1,206 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.is.configuration.toml.generator.util;
+
+import com.moandjiezana.toml.Toml;
+import com.moandjiezana.toml.TomlWriter;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.wso2.is.configuration.diff.creater.exception.ConfigMigrateException;
+import org.wso2.is.configuration.diff.creater.utils.MigrationConstants;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * The class which contains methods needed for process of toml generation.
+ */
+public class TomlGenerator {
+
+    private static final Logger log = LogManager.getLogger(TomlGenerator.class);
+
+    /**
+     * Write the configuration map to a file.
+     *
+     * @param tomlKeyValueMap map of configuration.
+     */
+    public void writeToTOMLFile(Map<String, Object> tomlKeyValueMap) {
+
+        TomlWriter writer = new TomlWriter();
+        try {
+            File outputDeploymentFile = getOutputTomlFile();
+            if (outputDeploymentFile.createNewFile()) {
+                writer.write(tomlKeyValueMap, outputDeploymentFile);
+            }
+        } catch (IOException e) {
+            log.error("error saving configuration", e);
+        }
+    }
+
+    private File getOutputTomlFile() {
+
+        File outputDeploymentFile = new File(TomlGeneratorConstants.UPDATED_DEPLOYMENT_TOML);
+        if (outputDeploymentFile.exists()) {
+            if (!outputDeploymentFile.delete()) {
```

Comment:
Shall we improve this logic as `if (outputDeploymentFile.exists() && !outputDeploymentFile.delete()) `